cmd/osbuild-worker/main.go

```diff
@@ -154,12 +157,21 @@ func main() {
 		}
 	}
 
+	var azureCredentials *OSBuildJobImplAzureCredentials
+	if config.Azure != nil {
+		azureCredentials, err = loadAzureCredentials(config.Azure.Credentials)
+		if err != nil {
+			log.Fatalf("cannot load azure credentials: %v", err)
+		}
+	}
+
 	jobImpls := map[string]JobImplementation{
 		"osbuild": &OSBuildJobImpl{
 			Store:        store,
 			Output:       output,
 			KojiServers:  kojiServers,
 			GCPCredsPath: config.GCP.Credentials,
+			AzureCreds:   azureCredentials,
```

Comment:
Any reason this is different from GCP? The Azure config is a pointer to a struct, while GCP is not, and the Azure config is parsed in `main()` whilst GCP is passed on as just the path to the creds.

Reply:
I added a comment to this: Basically, the fileformat for Azure credentials is created by us and I want to make sure to fail as soon as possible if it's malformed. We probably want to do something similar for GCP too. If we have wrong credentials, we would like to fail as soon as possible and not wait for the first job.

Reply:
Yeah, sounds reasonable to do also for GCP, although it would be just `[]byte`, not our own structure... For GCP, not providing the credentials can still work if the worker is not running on GCP instance with proper roles associated with it. So to exclude the possibility to fail late, we would need to try to authenticate when the worker is starting up (which we discussed with @ondrejbudai). But then again it would probably make sense to be able to configure supported features in worker's configuration, because now every worker would have to be able to authenticate with GCP.